src/parser/types.rs

```diff
@@ -399,7 +413,11 @@ impl Ty {
                     // otherwise assume it is a GenericArray
                     // [Field; NN]
                     //         ^^^
-                    (TokenKind::Generic(name), TokenKind::RightBracket) => {
+                    (TokenKind::Identifier(name), TokenKind::RightBracket) => {
+                        if !is_generic_parameter(name) {
```

Comment:
this is valid code but will error here:

```rust
const len = 3;

fn thing([Field; len]) { /* ... */ }
```

Reply:
right, we would need to also rely on the GenericArray to carry that constant variable.
I guess this is another reason why we need to refactor the `Array` to carry symbolic for both concrete Array and generic Array.

Reply:
(as you pointed out in the main PR)